### PR TITLE
Use mypyc to compile modules

### LINKED ISSUE #24: Pointer lag
With a Razer gaming mouse, moving the mouse results in significant lag (like the cursor will continue moving for a second after I've stopped). If I keep the mouse still and just click the button, there is no noticeable lag.

This suggests to me that (because it is a gaming mouse) the mouse is sending events faster than they are being processed, creating a blocked queue of events which can take seconds to finish being processed after incoming events have finished.

2 possible solutions I can think of:
 1. Improve processing performance to reduce the likelihood of queue buildup (this probably requires 1ms or under processing times).
 2. Figure out a way to discard some events or clear the event queue when a buildup happens.

### PR DESCRIPTION
**Pull request by Dreamsorcerer**
This compiles the modules which are involved in processing BT/HID events in order to increase performance.

Fixes #24

### CODE REVIEW

File: .mypy.ini [hunk 1]
```diff
@@ -0,0 +1,25 @@
+[mypy]
```
Comment (Dreamsorcerer) [inline .mypy.ini:1]: The basic idea here is to completely disallow all dynamic typing in these modules. This ensures mypyc has all the information needed to provide the best possible performance improvements.

File: .mypy.ini [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+files = a1314_message_filter.py, adapter.py, bluetooth_devices.py, hid_message_filter.py, mouse_message_filter.py
```
Comment (Dreamsorcerer) [inline .mypy.ini]: agent.py, compatibility_devices.py and hid_devices.py should be added back into this list, but currently they error when compiled with mypyc.

File: stubs/dasbus/connection.pyi [hunk 3]
```diff
@@ -0,0 +1,48 @@
+from typing import Callable, Dict, List, Optional
```
Comment (Dreamsorcerer) [inline stubs/dasbus/connection.pyi:1]: stubs/ just contains the minimal stubs needed to satisfy mypy and eliminate all dynamic typing. There are some cheats and such in here, but that shouldn't be an issue.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+name: CI
```
Comment (Dreamsorcerer) [inline .github/workflows/ci.yml:1]: I'm guessing this won't run on this repository until it's actually been merged initially.
You can see the CI run on my fork though: https://github.com/Dreamsorcerer/bthidhub/commits/master

**Comment by ruundii** [comment]
Hi Sam,

I won't be able to look at the pull request for the next few weeks, but I appreciate your contributions. I have invited you as a contributor to this repository, so feel free to merge your changes to the main branch

**Comment by Dreamsorcerer** [comment]
@ruundii Unfortunately, I've just realised it doesn't really resolve the lag issue after all (though may still be a good idea overall).

When I start the application, there is no noticeable lag. If I disconnect my laptop and reconnect it (from the laptop's BT settings), then often the lag appears after reconnecting. Restarting the app will often go back to being lag-free.
    Additionally, whenever I have the BT settings open on my laptop (Ubuntu) there is a very high level of lag. As if the laptop's scanning slows down the RPi's processing time or something (but, this doesn't happen when the mouse is connected directly).

Any ideas what might be causing this lag after reconnection?

I've tried profiling the app, but I don't see any major differences. Maybe there is more time spent in asyncio_glib/glib_selector.py, but I'm not too convinced the problem is there, it might be an issue not exposed by the profiler.

**Comment by ruundii** [comment]
from when i was developing and profiling it I remember few things which may cause issues
1) scanning. as far as I remember when you open BT in ubuntu it does scanning, basically, BT slows down then. It is just a thing in BT protocol, basically, it needs to send/receive many messages over limited number of frequencies etc. so it is bound to slow down. At some point I want to check if it would be better to utilise HoG/GATT for the bthidhub, possibly this will solve some of such issues, but it is a big thing to try out. (e.g. see https://github.com/HeadHodge/Bluez-HID-over-GATT-Keyboard-Emulator-Example/blob/main/gattServer.py)

2) there is master/slave role in Bluetooth. basically out of two connected devices one is 'controlling the clock'. it is very important for RPi to be in master role for the BT connection with the host PC, otherwise it is bound to lag. bluetooth_devices.py sets this master role. I would check the state of master role when lag is occurring to ensure it is set

3) Wifi. Unfortunately I found sometimes massive interference between Bluetooth and Wifi.

**Comment by Dreamsorcerer** [comment]
I think you've got it. It appears that it sets to master on startup, but doesn't do it again when the device reconnects. I'll have a look tomorrow and create a PR to fix it.

**Comment by Dreamsorcerer** [comment]
Nevermind, that only took 5 mins. :P

**Comment by Dreamsorcerer** [comment]
I'm happy to merge the other PRs, but I think this one is probably best if tested by someone other than me first.

Also, when you do have some time, it'd be a great help if you could take a look at #21. This is the last issue causing me major problems, and the problem appears to be in the forked bluez, which is not something I know how to debug.

**Pull request merged**